testWork/testWork1-2/testWork1-2/main.c

```diff
@@ -0,0 +1,73 @@
+#include <stdio.h>
+#include <stdbool.h>
+
+void shellSort(int* array, const int length)
+{
+    for (int step = length / 2; step > 0; step /= 2)
+        for (int i = step; i < length; ++i)
+        {
+            int j = i;
+            int temp = array[i];
+            for (j; j >= step; j -= step)
```

Comment:
```suggestion
            for (; j >= step; j -= step)
```
j тут буквально ничего не делает, с тем же успехом вместо неё можно 123 написать.